### ISSUE TITLE
StyleBoxTexture behaves oddly when content margins are smaller than texture margins

### ISSUE THREAD

**Issue by ctkerns** [godotengine/godot#73468]
### Godot version

v4.0.rc2.official [d2699dc7a]

### System information

Windows 10, Vulkan

### Issue description

Since https://github.com/godotengine/godot/pull/72124 I have noted some weird behavior in my converted projects from Godot 3.5

The GUI System seems incapable of scaling the borders of a StyleBoxTexture beyond their native resolution. For the sake of demonstration, I have created a simple 9 patch texture to demonstrate this. The texture is 128x128 with each edge being 19 pixels wide.

When applied to a button, as shown, with -1 for each content margin, the margins default to 19 pixels as shown in the image below.

![Capture](https://user-images.githubusercontent.com/46550204/219510953-41e2a323-f494-48f1-87ad-6b8af7f9ac54.PNG)

This appears reasonable, but in some cases, you might not want such a large margin. When reducing the content margins to 8 pixels as shown in the image below, there is an obvious issue. The borders now overlap the center content.

![Screenshot 2023-02-16 175355](https://user-images.githubusercontent.com/46550204/219511797-5c1d6649-b01e-416d-9ee2-10efefd39988.png)

Reducing further, the borders will now even overlap with themselves, which is very jarring.

![Screenshot 2023-02-16 175454](https://user-images.githubusercontent.com/46550204/219512010-84396bd0-f1c2-45de-8afd-54c53e1e8567.png)

I haven't had this issue in Godot 3.5, where the borders tend to scale down as required. The pixel art example is fairly trivial, but in my case, I use high quality assets with a high resolution, and this new behavior now makes it impossible to use these assets properly in any context.

For my purposes, when converting to Godot 4, I will either have to recreate these assets so that the borders are sized correctly depending on the context while also reducing their fidelity, or I would have to use the expand margins which throws off the look and feel of my UI.

Ideally, I think, StyleBoxTextures would have better options that would allow for more control over how these borders are scaled, or alternatively, the borders will be scaled in a way that does not produce such jarring results.

Some related issues:  
#71913  
#72147

### Steps to reproduce

1. Open attached project  
2. Open "rainbow.tres"
3. Adjust content margins

### Minimal reproduction project

[StyleBoxTexture.zip](https://github.com/godotengine/godot/files/10761601/StyleBoxTexture.zip)

**Comment by YuriSizov** [comment]
What exactly is the expected look here?

**Comment by ctkerns** [comment]
So in 3.5.1, this is what you get with the exact same setup, using 8 pixel content margins. Compare this to the second image above. In this version, the borders do not overlap each other, the content is not covered by the border, and the borders are downscaled.

![Screenshot 2023-02-16 210835](https://user-images.githubusercontent.com/46550204/219531445-c99f12f2-4cfd-4215-9aa3-59fad7105e57.png)

I have no clue how this is supposed to behave, but the 3.5 behavior suited my needs far better than this new behavior.

3.5.1 project for convenience:
[StyleBoxTexture.zip](https://github.com/godotengine/godot/files/10762599/StyleBoxTexture.zip)

**Comment by YuriSizov** [comment]
Maybe I'm doing something wrong, but backporting your reproduction project down to beta 1 yields the same exact look, borders do not get squished. I will try to investigate it more, but it doesn't look related to the recent changes at all. Perhaps something changed about how we handle 9-patch?

Looking at the 3.x version, I'm not confident that the way it looks is what you should expect it to look. As in, I don't see anything in the configuration of the resource that would tell Godot to squish the borders. Even if this is the look that you want here in particular.

**Comment by ctkerns** [comment]
Yeah, I'm not necessarily confident the 3.x behavior is how it should work. If it is a simple matter that content margins should be greater than or equal to texture margins, and all other behavior is unsupported, given that the borders always render in their native resolution, then the 4.0 behavior seems fairly consistent. Though this constraint is not clear from the user's perspective.

In my case, I developed assets expecting the 3.x behavior that now don't display the same way. And maybe it is the best solution for the engine going forward that it behaves in the new way at my expense; I can accept that.

I still think there should be some separation between the native resolution of the border and the display size. After all, you can pull 9 patch textures from the web into your project that don't anticipate the resolution of your project, and thus display differently than intended. That's just my humble opinion though, I'd like to see how this change affects other people's games.

**Comment by Mickeon** [comment]
I can reproduce this... issue(?) in 4.3 dev still.
Not sure. I feel like, if anything, the downscaled borders feel worse, especially for pixel-art games. I do have a few memories of being annoyed at the downscaling back in 3.x, but I can see why one would want the old behavior back on _some_ occasions? Even then, it's not exactly pretty in most scenarios...
